api/internal/git/repospec.go

```diff
@@ -110,12 +110,20 @@ const (
 // the parts.
 func parseGitURL(n string) (
 	host string, orgRepo string, path string, gitRef string, gitSubmodules bool, gitSuff string, gitTimeout time.Duration) {
+	// parse query first
```

Comment:
(Not for this PR, but maybe a future refactoring one) 7 return values is so many... we really should make this into a struct at some point.

Reply:
Created an issue: #4866